Compiling develop branch with intel2022a

@TomMelt @einola 
When compiling the develop branch on the HPC fram sigma2, I get the following error: 

In file included from /cluster/home/guibou/src/nextsimdg/core/src/ModelMPI.cpp:6:
/cluster/home/guibou/src/nextsimdg/core/src/include/ModelMPI.hpp:30:57: error: could not convert ‘nullptr’ from ‘std::nullptr_t’ to ‘MPI_Comm’ {aka ‘int’}
   30 |     inline static ModelMPI& getInstance(MPI_Comm comm = nullptr)
      |                                                         ^~~~~~~
      |                                                         |
      |                                                         std::nullptr_t
make[2]: *** [CMakeFiles/nextsimlib.dir/build.make:76: CMakeFiles/nextsimlib.dir/core/src/ModelMPI.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:257: CMakeFiles/nextsimlib.dir/all] Error 2
make: *** [Makefile:91: all] Error 2


Here is my cmake command:

cmake ..  -DCMAKE_BUILD_TYPE=Release -DWITH_THREADS=ON -DPYTHON_EXECUTABLE=~/.conda/envs/pynex/bin/python -DENABLE_MPI=ON -DENABLE_OASIS=ON -DOASIS_DIR=/cluster/projects/nn9878k/guibou/NANUK/oa3-mct_iimpi2022a/  -DBUILD_TESTS=OFF -DDynamicsType=DG1

(Removing OASIS doesn't change anything)

My current commit is: commit 8e85310c35453bf624417ffa889e843fa6942a1b

It used to work fine using the issue459_oasis_ocean_coupling branch and (at least) commit 37b40fe384cd45db29de4184185a01df0cf638c7


I use intel compilers: impi/2021.6.0-intel-compilers-2022.1.0

--- a/core/src/include/ModelMPI.hpp
+++ b/core/src/include/ModelMPI.hpp
@@ -27,7 +27,7 @@ private:
     static bool doOnce();
 
 public:
-    inline static ModelMPI& getInstance(MPI_Comm comm = nullptr)
+    inline static ModelMPI& getInstance(MPI_Comm comm = MPI_COMM_NULL)
     {
         static ModelMPI instance = ModelMPI(comm);
         if (instance.isInitialized) {

